internal/controller/bmcversionset_controller_test.go

```diff
@@ -19,384 +21,488 @@ import (
+			},
+		}
+		Eventually(Get(bmcVersion01)).Should(Succeed())
+
+		By("Ensuring that the BMCVersion has been created")
+		bmcVersion02 := &metalv1alpha1.BMCVersion{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: bmcVersionList.Items[1].Name,
+			},
+		}
+		Eventually(Get(bmcVersion02)).Should(Succeed())
+
+		By("Checking if the status has been updated")
+		Eventually(Object(bmcVersionSet)).Should(SatisfyAll(
+			HaveField("Status.FullyLabeledBMCs", BeNumerically("==", 2)),
+			HaveField("Status.AvailableBMCVersion", BeNumerically("==", 2)),
+			HaveField("Status.FailedBMCVersion", BeNumerically("==", 0)),
+		))
+
+		By("Checking the bmcVersion01 have completed")
+		Eventually(Object(bmcVersion01)).Should(SatisfyAll(
+			HaveField("Status.State", metalv1alpha1.BMCVersionStateCompleted),
+			HaveField("Spec.Version", bmcVersionSet.Spec.BMCVersionTemplate.Version),
+			HaveField("Spec.Image.URI", bmcVersionSet.Spec.BMCVersionTemplate.Image.URI),
+			HaveField("OwnerReferences", ContainElement(metav1.OwnerReference{
+				APIVersion:         "metal.ironcore.dev/v1alpha1",
+				Kind:               "BMCVersionSet",
+				Name:               bmcVersionSet.Name,
+				UID:                bmcVersionSet.UID,
+				Controller:         ptr.To(true),
+				BlockOwnerDeletion: ptr.To(true),
+			})),
+		))
+
+		By("Checking the bmcVersion02 have completed")
+		Eventually(Object(bmcVersion02)).Should(SatisfyAll(
+			HaveField("Status.State", metalv1alpha1.BMCVersionStateCompleted),
+			HaveField("Spec.Version", bmcVersionSet.Spec.BMCVersionTemplate.Version),
+			HaveField("Spec.Image.URI", bmcVersionSet.Spec.BMCVersionTemplate.Image.URI),
+			HaveField("OwnerReferences", ContainElement(metav1.OwnerReference{
+				APIVersion:         "metal.ironcore.dev/v1alpha1",
+				Kind:               "BMCVersionSet",
+				Name:               bmcVersionSet.Name,
+				UID:                bmcVersionSet.UID,
+				Controller:         ptr.To(true),
+				BlockOwnerDeletion: ptr.To(true),
+			})),
+		))
+
+		By("Checking if the status has been updated")
+		Eventually(Object(bmcVersionSet)).Should(SatisfyAll(
+			HaveField("Status.FullyLabeledBMCs", BeNumerically("==", 2)),
+			HaveField("Status.AvailableBMCVersion", BeNumerically("==", 2)),
+			HaveField("Status.CompletedBMCVersion", BeNumerically("==", 2)),
+			HaveField("Status.InProgressBMCVersion", BeNumerically("==", 0)),
+			HaveField("Status.FailedBMCVersion", BeNumerically("==", 0)),
+		))
+
+		By("Deleting the resource")
+		Expect(k8sClient.Delete(ctx, bmcVersionSet)).To(Succeed())
+
+		// cleanup
+		Expect(k8sClient.Delete(ctx, bmcVersion01)).To(Succeed())
+		Expect(k8sClient.Delete(ctx, bmcVersion02)).To(Succeed())
+
+		maintenanceList := &metalv1alpha1.ServerMaintenanceList{}
+		Eventually(List(maintenanceList)).Should(Succeed())
+		for _, maintenance := range maintenanceList.Items {
+			Expect(k8sClient.Delete(ctx, &maintenance)).To(Succeed())
+		}
+
+		server01 := &metalv1alpha1.Server{
```

Comment:
I would argue that the BMC and Server cleanup could be moved in the `AfterEach` block for deduplication, because the BMCs are setup in the `BeforeEach` block.